api/src/server.rs

```diff
@@ -95,17 +102,25 @@ where
 }
 
 #[derive(Clone)]
-pub struct Server<C, R> {
+pub struct Server<C, IR, MR> {
+    peer_id: PeerId,
     network: Network,
-    recon: R,
+    interest: IR,
+    model: MR,
     marker: PhantomData<C>,
 }
 
-impl<C, R: Recon> Server<C, R> {
-    pub fn new(network: Network, recon: R) -> Self {
+impl<C, IR, MR> Server<C, IR, MR>
+where
+    IR: Recon<Key = Interest>,
```

Comment:
Why `IR` and `MR`? Guessing it's for "InterestRecon" and "ModelRecon". Maybe just use `I` and `M` or `Interest` and `Model`?